app/models/vm_scan.rb

```diff
@@ -102,6 +102,8 @@ def call_snapshot_create
       if vm.kind_of?(ManageIQ::Providers::Openstack::CloudManager::Vm) ||
          vm.kind_of?(ManageIQ::Providers::Microsoft::InfraManager::Vm)
         return unless create_snapshot(vm)
+      elsif vm.kind_of?(ManageIQ::Providers::Azure::CloudManager::Vm) && vm.require_snapshot_for_scan?
```

Comment:
@chessbyte I agree.  It does open up a larger can of worms because there is similar logic for the snapshot removal and it seems that there is VMware logic built in to the code here in both cases.  Doing some cleanup in this file would be helpful...